File: leetcode-04/solutions/python3/0128.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        ans = 0
        set_ = set(nums)

        for num in set_:
            if num - 1 not in set_:
                currNum = num
                currAns = 1
                while currNum + 1 in set_:
                    currNum += 1
                    currAns += 1
                ans = max(ans, currAns)

        return ans

File: leetcode-04/solutions/cpp/0128.cpp
class Solution {
 public:
  int longestConsecutive(vector<int>& nums) {
    int ans = 0;
    unordered_set<int> set(nums.begin(), nums.end());

    for (int num : set)
      if (!set.count(num - 1)) {
        int currNum = num;
        int currAns = 1;
        while (set.count(currNum + 1)) {
          currNum++;
          currAns++;
        }
        ans = max(ans, currAns);
      }

    return ans;
  }
};